app/models/search.rb

```diff
@@ -0,0 +1,108 @@
+class Search
+  include ActiveModel::Model
+
+  attr_accessor :last_name
+  attr_reader :date_of_birth
+
+  validates :last_name, presence: true
+  validate :date_of_birth_is_valid
+
+  def date_of_birth=(date_fields)
+    return if date_fields.nil?
+
+    date_fields.map! { |f| f.is_a?(String) ? word_to_number(f.strip).to_s : f }
+
+    month_is_a_word = date_fields[1]&.to_i&.zero? && date_fields[1].length.positive?
+    date_fields[1] = word_to_month_number(date_fields[1]) if month_is_a_word
+
+    @date_of_birth = Struct.new(:year, :month, :day).new(*date_fields)
```

Comment:
Search is currently broken because we pass the `to_s` of this struct to the API client. Any reason this can't be a `Date`?

Needs a manual test to confirm fix, system spec didn't pick this up. 

I think it's worth adding to the spec for this method an expectation that checks the result of calling `to_s` on the resulting `date_of_birth`.